cfme/infrastructure/virtual_machines.py

```diff
@@ -86,46 +87,169 @@
+    including_entities = View.include(VMEntities, use_parent=True)
+
+    def reset_page(self):
+        self.entities.search.clear_search()
+
+
+class VmTemplatesAllForProviderView(InfraVMView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_infra_vms and
+            str(self.entities.title.text) ==
+            'VM or Templates under Provider \"{}\"'.format(self.context['object'].provider.name))
+    toolbar = View.nested(VMToolbar)
+    sidebar = View.nested(VmsTemplatesAccordion)
+    including_entities = View.include(VMEntities, use_parent=True)
+
+    def reset_page(self):
+        self.entities.search.clear_search()
+
+
+class VmsOnlyAllView(InfraVMView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_infra_vms and
+            self.sidebar.vms.tree.currently_selected == 'All VMs' and
+            self.entities.title.text == 'All VMs')
+    toolbar = View.nested(VMToolbar)
+    sidebar = View.nested(VmsTemplatesAccordion)
+    including_entities = View.include(VMEntities, use_parent=True)
+
+    def reset_page(self):
+        self.entities.search.clear_search()
+
+
+class TemplatesOnlyAllView(InfraVMView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_infra_vms and
+            self.sidebar.templates.tree.currently_selected == 'All Templates' and
+            self.entities.title.text == 'All Templates')
+    toolbar = View.nested(VMToolbar)
+    sidebar = View.nested(VmsTemplatesAccordion)
+    including_entities = View.include(VMEntities, use_parent=True)
+
+
+class InfraVmSummaryView(VMDetailsEntities):
+    operating_ranges = SummaryTable(title="Normal Operating Ranges (over 30 days)")
+    datastore_allocation = SummaryTable(title="Datastore Allocation Summary")
+    datastore_usage = SummaryTable(title="Datastore Actual Usage Summary")
+
+
+class InfraVmDetailsView(InfraVMView):
+    title = Text('#explorer_title_text')
+    toolbar = ConditionalSwitchableView(reference='entities.title')
+
+    @toolbar.register(lambda title: "VM and Instance" in title)
+    class VmsToolbar(InfraVmDetailsToolbar):
+        pass
+
+    @toolbar.register(lambda title: "VM Template and Image" in title)
+    class TemplatesToolbar(InfraVmTemplatesGenericDetailsToolbar):
+        pass
+    sidebar = View.nested(VmsTemplatesAccordion)
+    entities = View.nested(InfraVmSummaryView)
+
+    @property
+    def is_displayed(self):
+        expected_name = self.context['object'].name
+        expected_provider = self.context['object'].provider.name
+        try:
+            relationship_provider_name = self.entities.relationships.get_text_of('Infrastructure '
+                                                                                 'Provider')
+        except NameError:
+            if self.sidebar.vmstemplates.tree.currently_selected[-1] in ['<Archived>',
+                                                                         '<Orphaned>']:
```

Comment:
pull the currently_selected value into an intermediate variable to remove the need of multilining the if, it will look better.